public/plugin.tsx

```diff
@@ -329,7 +333,24 @@ export class AssistantPlugin
         getDisplayName: () => 'Generate visualization',
         getIconType: () => 'visLine' as const,
         execute: async () => {
-          core.application.navigateToApp(TEXT2VIZ_APP_ID);
+          const dataset = data.query.queryString.getQuery().dataset;
+          const url = new URL(core.application.getUrlForApp(TEXT2VIZ_APP_ID, { absolute: true }));
+          if (dataset && dataset.type === 'INDEX_PATTERN') {
+            url.searchParams.set(INDEX_PATTERN_URL_SEARCH_KEY, dataset.id);
```

Comment:
Don't we need the data source info?

Reply:
Not for t2viz at the moment, because it will request the index pattern details by id, and retrieve data source id from there.